libs/ui-components/src/components/modals/massModals/MassApproveDeviceModal/MassApproveDeviceModal.tsx

```diff
@@ -50,24 +50,29 @@ const MassApproveDeviceModal: React.FC<MassApproveDeviceModalProps> = ({
   const [errors, setErrors] = React.useState<string[]>();
   const {
     user,
-    fetch: { post },
+    fetch: { post, patch },
   } = useAppContext();
 
   const approveEnrollments = async (values: MassApproveDeviceFormValues) => {
     setProgress(0);
     setErrors(undefined);
     const promises = pendingEnrollments.map(async (r, index) => {
       const labels = toAPILabel(values.labels);
-      const aliasLabel = templateToName(index, values.deviceAlias);
-      if (aliasLabel) {
-        labels.alias = aliasLabel;
-      }
+      const deviceAlias = templateToName(index, values.deviceAlias);
 
       await post<EnrollmentRequestApproval>(`enrollmentrequests/${r.metadata.name}/approval`, {
         approved: true,
         labels,
         approvedBy: user,
       });
+      // Workaround until EDM-607 allows us to specify the alias in the approval request
+      await patch<DeviceSpec>(`devices/${r.metadata.name}`, [
```

Comment:
Device alias will be finally kept as a label. So when the API is modified, I'll make the necessary adjustments to rollback these changes.